### PR TITLE
Replace deprecated `datetime.utcnow()` w/ a tz-aware variant

### PR DESCRIPTION
**Pull request by radez**
Calling `datetime.datetime.utcnow()` emits a deprecation warning now.
Python 3.12 reports:
```python-traceback
DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
```
This patch replaces it with a non-deprecated cross-python implementation.

**What kind of change does this PR introduce?**
  - [x] refactoring

**Checklist**:

  - [x] I think the code is well written
  - [x] I wrote [good commit messages][1]
  - [x] I have [squashed related commits together][2] after the changes have been approved
  - [x] I used the same coding conventions as the rest of the project
  - [x] The new code doesn't generate linter offenses
  - [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences

[1]: http://chris.beams.io/posts/git-commit/
[2]: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

### CODE REVIEW

File: cherrypy/test/sessiondemo.py [hunk 1]
```diff
@@ -123,9 +123,11 @@ def page(self):
             'reqcookie': cherrypy.request.cookie.output(),
             'sessiondata': list(cherrypy.session.items()),
             'servertime': (
-                datetime.utcnow().strftime('%Y/%m/%d %H:%M') + ' UTC'
+                datetime.datetime.now(
+                    datetime.UTC).strftime('%Y/%m/%d %H:%M') + ' UTC'
             ),
-            'serverunixtime': calendar.timegm(datetime.utcnow().timetuple()),
+            'serverunixtime': calendar.timegm(
+                datetime.datetime.utcnow(datetime.UTC).timetuple()),
```
Comment (webknjaz) [inline cherrypy/test/sessiondemo.py]: (minor nitpick)
```suggestion
            'serverunixtime':
            calendar.timegm(
                datetime.datetime.utcnow(datetime.UTC).timetuple(),
            ),
```

File: cherrypy/test/sessiondemo.py [hunk 2]
```diff
@@ -123,9 +123,11 @@ def page(self):
             'reqcookie': cherrypy.request.cookie.output(),
             'sessiondata': list(cherrypy.session.items()),
             'servertime': (
-                datetime.utcnow().strftime('%Y/%m/%d %H:%M') + ' UTC'
+                datetime.datetime.now(
+                    datetime.UTC).strftime('%Y/%m/%d %H:%M') + ' UTC'
```
Comment (webknjaz) [inline cherrypy/test/sessiondemo.py]: It occurred to me that "UTC" could be a part of the actual string template. So why not embed it?
```suggestion
                datetime.datetime.now(datetime.UTC).
                strftime('%Y/%m/%d %H:%M UTC')
```

File: cherrypy/test/sessiondemo.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 """A session demonstration app."""
 
 import calendar
-from datetime import datetime
+import datetime
```
Comment (webknjaz) [inline cherrypy/test/sessiondemo.py]: I think, I'd prefer a scoped "import from" variant, because reading things like `datetime.datetime` is rather confusing.
```suggestion
from datetime import datetime, UTC
```

https://youtu.be/x-kB2o8sd5c?t=1038&feature=shared nicely explains when this might be better.

File: cherrypy/test/test_logging.py [hunk 4]
```diff
@@ -216,7 +216,7 @@ def test_utc_in_timez(monkeypatch):
 
     class mock_datetime:
         @classmethod
-        def utcnow(cls):
+        def now(cls, _):
```
Comment (webknjaz) [inline cherrypy/test/test_logging.py]: Could you still name the argument, so the signature matches?
```suggestion
        def now(cls, tz):
```

If flake8 is unhappy, that would warrant an inline `# noqa: <error-code>` with a code comment explaining it.

File: cherrypy/test/test_logging.py [hunk 5]
```diff
@@ -199,7 +199,7 @@ def test_custom_log_format(log_tracker, monkeypatch, server):
 
 
 def test_utc_in_timez(monkeypatch):
-    """Test that ``LazyRfc3339UtcTime`` is rendered as ``str`` using UTC timestamp."""
+    """Test ``LazyRfc3339UtcTime`` renders as ``str`` UTC timestamp."""
```
Comment (webknjaz) [inline cherrypy/test/test_logging.py]: I just realized that this is also changed in #2008 which would be conflicting.

While not necessarily related to this PR, I think it's fine to keep it at a part of this patch and won't demand a dedicated PR.

File: .readthedocs.yml
Comment (webknjaz) [inline .readthedocs.yml:1]: The changes to this RTD configuration file are definitely not related to changing the deprecated function calls, so this should be submitted as a separate PR.

Comment (radez) [reply .readthedocs.yml:1]: Agreed, this shouldn't have been in here. I'll fix this.

File: cherrypy/test/test_logging.py (hunk 5 above)
Comment (radez) [reply cherrypy/test/test_logging.py]: I just pulled this out of 2008. I'll let this make the update instead since it's relevant.

File: cherrypy/test/test_logging.py (hunk 4 above)
Comment (radez) [reply cherrypy/test/test_logging.py]: Done

File: cherrypy/test/sessiondemo.py (hunk 3 above)
Comment (radez) [reply cherrypy/test/sessiondemo.py]: done

File: cherrypy/test/sessiondemo.py (hunk 2 above)
Comment (radez) [reply cherrypy/test/sessiondemo.py]: done

File: cherrypy/test/sessiondemo.py (hunk 1 above)
Comment (radez) [reply cherrypy/test/sessiondemo.py]: done

**Comment by webknjaz** [comment]
I have clicked the "Rebase" button via GH UI.

File: cherrypy/test/sessiondemo.py [hunk 6]
```diff
@@ -123,9 +123,12 @@ def page(self):
             'reqcookie': cherrypy.request.cookie.output(),
             'sessiondata': list(cherrypy.session.items()),
             'servertime': (
-                datetime.utcnow().strftime('%Y/%m/%d %H:%M') + ' UTC'
+                datetime.now(UTC).strftime('%Y/%m/%d %H:%M UTC')
+            ),
+            'serverunixtime':
+            calendar.timegm(
+                datetime.datetime.utcnow(UTC).timetuple()
```
Comment (webknjaz) [inline cherrypy/test/sessiondemo.py]: Looks like this one place has the nested datetime forgotten:
```suggestion
                datetime.utcnow(UTC).timetuple(),
```

File: cherrypy/test/test_logging.py (hunk 4 above)
Comment (webknjaz) [reply cherrypy/test/test_logging.py]: Thanks! Marking this as resolved!

File: cherrypy/test/sessiondemo.py (hunk 3 above)
Comment (webknjaz) [reply cherrypy/test/sessiondemo.py]: Thanks! Marking this as resolved!

File: cherrypy/test/sessiondemo.py (hunk 2 above)
Comment (webknjaz) [reply cherrypy/test/sessiondemo.py]: Thanks! Marking this as resolved!

File: cherrypy/test/sessiondemo.py (hunk 1 above)
Comment (webknjaz) [reply cherrypy/test/sessiondemo.py]: Thanks! Marking this as resolved!

File: cherrypy/test/sessiondemo.py [hunk 7]
```diff
@@ -2,7 +2,7 @@
 """A session demonstration app."""
 
 import calendar
-from datetime import datetime
+from datetime import datetime, UTC
```
Comment (webknjaz) [inline cherrypy/test/sessiondemo.py]: Looks like `UTC` needs to be accessed differently:
`E   ImportError: cannot import name 'UTC'`

Comment (webknjaz) [reply cherrypy/test/sessiondemo.py]: At least under Python 3.6...

Comment (webknjaz) [reply cherrypy/test/sessiondemo.py]:
```suggestion
from datetime import datetime

try:
    from datetime import UTC  # Python 3.11+
except ImportError:
    # Python 3.6-3.10
    from datetime import timezone as _timezone
    UTC = _timezone.utc
    del _timezone
```

File: cherrypy/_cplogging.py [hunk 8]
```diff
@@ -452,6 +452,6 @@ def emit(self, record):
 
 class LazyRfc3339UtcTime(object):
     def __str__(self):
-        """Return utcnow() in RFC3339 UTC Format."""
-        iso_formatted_now = datetime.datetime.utcnow().isoformat('T')
+        """Return datetime in RFC3339 UTC Format."""
+        iso_formatted_now = datetime.datetime.now(datetime.UTC).isoformat('T')
```
Comment (webknjaz) [inline cherrypy/_cplogging.py]:
```suggestion
        iso_formatted_now = datetime.datetime.now(
            datetime.timezone.utc,
        ).isoformat('T')
```

File: cherrypy/lib/locking.py [hunk 9]
```diff
@@ -19,10 +19,10 @@ def after(cls, elapsed):
         """
         Return a timer that will expire after `elapsed` passes.
         """
-        return cls(datetime.datetime.utcnow() + elapsed)
+        return cls(datetime.datetime.now(datetime.UTC) + elapsed)
```
Comment (webknjaz) [inline cherrypy/lib/locking.py]:
```suggestion
        return cls(
            datetime.datetime.now(datetime.timezone.utc) + elapsed,
        )
```

File: cherrypy/lib/locking.py [hunk 10]
```diff
@@ -19,10 +19,10 @@ def after(cls, elapsed):
...
 
     def expired(self):
-        return datetime.datetime.utcnow() >= self.expiration
+        return datetime.datetime.now(datetime.UTC) >= self.expiration
```
Comment (webknjaz) [inline cherrypy/lib/locking.py]:
```suggestion
        return datetime.datetime.now(
            datetime.timezone.utc,
        ) >= self.expiration
```

File: cherrypy/test/sessiondemo.py [hunk 11]
```diff
@@ -3,6 +3,14 @@
...
 import calendar
 from datetime import datetime
+
+try:
+    from datetime import UTC  # Python 3.11+
+except ImportError:
+    # Python 3.6-3.10
+    from datetime import timezone as _timezone
+    UTC = _timezone.utc
+    del _timezone
```
Comment (webknjaz) [inline cherrypy/test/sessiondemo.py]:
```suggestion
from datetime import timezone as _timezone
```

File: cherrypy/test/sessiondemo.py [hunk 12]
```diff
@@ -123,9 +131,12 @@ def page(self):
             'reqcookie': cherrypy.request.cookie.output(),
             'sessiondata': list(cherrypy.session.items()),
             'servertime': (
-                datetime.utcnow().strftime('%Y/%m/%d %H:%M') + ' UTC'
+                datetime.now(UTC).strftime('%Y/%m/%d %H:%M UTC')
```
Comment (webknjaz) [inline cherrypy/test/sessiondemo.py]:
```suggestion
                datetime.now(_timezone.utc).strftime('%Y/%m/%d %H:%M UTC')
```

File: cherrypy/test/sessiondemo.py [hunk 13]
```diff
@@ -123,9 +131,12 @@ def page(self):
...
+            ),
+            'serverunixtime':
+            calendar.timegm(
+                datetime.utcnow(UTC).timetuple(),
```
Comment (webknjaz) [inline cherrypy/test/sessiondemo.py]:
```suggestion
                datetime.utcnow(_timezone.utc).timetuple(),
```

**Review by webknjaz — APPROVED** [review]
